gtsfm/utils/metrics.py

```diff
@@ -215,3 +223,36 @@ def compute_pose_errors(gt_wTi_list: List[Pose3], wTi_list: List[Pose3]) -> Dict
     metrics["rotation_angle_deg_errors"] = compute_rotation_angle_metrics(wRi_list, gt_wRi_list)
     metrics["translation_distance_errors"] = compute_translation_distance_metrics(wti_list, gt_wti_list)
     return metrics
+
+
+def log_sfm_summary() -> None:
+    """Dump to stdout a summary of metrics about the SfM reconstruction process."""
+    frontend_full_metrics_fpath = REPO_ROOT / "result_metrics" / "frontend_full.json"
+    frontend_metrics = io_utils.read_json_file(frontend_full_metrics_fpath)
+
+    rot_errs_deg = [
+        pair_stats["rotation_angular_error"] for pair_stats in frontend_metrics if pair_stats["rotation_angular_error"]
+    ]
+    trans_errs_deg = [
+        pair_stats["translation_angular_error"]
+        for pair_stats in frontend_metrics
+        if pair_stats["translation_angular_error"]
+    ]
+
+    logger.info("=============> Metrics report ==============>")
+    logger.info("Front-end median_rot_err_deg: %.2f", np.median(rot_errs_deg))
+    logger.info("Front-end max_rot_err_deg: %.2f", max(rot_errs_deg))
+
+    logger.info("Front-end median_trans_err_deg: %.2f", np.median(trans_errs_deg))
+    logger.info("Front-end max_trans_err_deg: %.2f", max(trans_errs_deg))
+
+    averaging_metrics_fpath = REPO_ROOT / "result_metrics" / "multiview_optimizer_metrics.json"
+    averaging_metrics = io_utils.read_json_file(averaging_metrics_fpath)
+
+    logger.info("Averaging median_rot_err_deg: %.2f", averaging_metrics["rotation_averaging_angle_deg"]["median_error"])
+    logger.info("Averaging max_rot_err_deg: %.2f", averaging_metrics["rotation_averaging_angle_deg"]["max_error"])
+
+    logger.info(
+        "Averaging median_trans_dist_err: %.2f", averaging_metrics["translation_averaging_distance"]["median_error"]
+    )
+    logger.info("Averaging max_trans_dist_err: %.2f", averaging_metrics["translation_averaging_distance"]["max_error"])
```

Comment:
I cant stress enough how hardcoding the metrics names here is so hacky. If we change the metric in a module we have to change all this.

Reply:
So you think we should just loop over the keys of the dictionary and print them? in other words, should the organization of the metric in the dictionary be structured so that we can print any dictionary?

Reply:
So some or the key value pairs are very verbose and not informative. I prefer in a future PR we come up with a more long term solution for printing out the summary statistics of interest with a generic metric class